tests/python/tir-transform/test_tir_transform_simplify.py

```diff
@@ -1325,6 +1326,34 @@ def expected(A: T.Buffer(16, "int32")):
                 A[i] = 42
 
 
+class TestAssume(BaseBeforeAfter):
+    """An assumption about buffer contents may apply to only part of a buffer
+    Like TestSimplifyUsingPartiallyKnownBufferConditional, but the
+    conditional is expressed as part of T.assume, instead of in the
+    control flow.
+    """
+
+    propagate_knowns_to_simplify_expressions = True
+
+    def before(A: T.Buffer(16, "int32")):
+        for i in T.serial(16):
+            T.evaluate(T.assume(i < 14 or A[i] == 0))
+
+        for i in T.serial(16):
+            if i < 14:
+                if A[i] == 0:
+                    A[i] = 42
+
+    def expected(A: T.Buffer(16, "int32")):
```

Comment:
Nitpick: Instead of duplicating the same function definition, I'd recommend `expected = before`.  That makes it clear to a reader that the simplification shouldn't cause any changes.